### PR TITLE
Metrodr e2e

### PR DESCRIPTION
**Pull request by raghavendra-talur**

### CODE REVIEW

File: hack/ocm-minikube.sh [hunk 1]
```diff
@@ -944,16 +944,18 @@ hub_kubeconfig_file_create()
 	kubectl --context $hub_cluster_name config view --flatten --minify >$hub_kubeconfig_file_path_name
 }
 exit_stack_push unset -f hub_kubeconfig_file_create
-deploy()
+ocm_minikube_deploy()
 {
+	hub_cluster_name=$1
+	spoke_cluster_names=$2
```
Comment (nirs) [inline hack/ocm-minikube.sh:950]: This creates a globals - is this used to pass the arguments to the functions below?

This is not a good direction - this code is has enough issues without globals.

File: hack/metrodr-setup.sh [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+	minikube_start "${MANAGEDCLUSTER1}"
+	minikube_start "${CEPHCLUSTER}"
+}
+
+function metrodr_external_ceph_cluster_create() {
+        kubectl --context "${CEPHCLUSTER}" create -f https://raw.githubusercontent.com/rook/rook/master/deploy/examples/common.yaml
+        kubectl --context "${CEPHCLUSTER}" create -f https://raw.githubusercontent.com/rook/rook/master/deploy/examples/crds.yaml
+        kubectl --context "${CEPHCLUSTER}" create -f https://raw.githubusercontent.com/rook/rook/master/deploy/examples/operator.yaml
+	#echo "======Deploying cluster=======" && sleep 100
+	kubectl --context "${CEPHCLUSTER}" create -f "${basedir}/dev-rook-cluster.yaml"
```
Comment (nirs) [inline hack/metrodr-setup.sh]: Inconsistent tabs and spaces.

File: hack/metrodr-setup.sh [hunk 3]
```diff
@@ -0,0 +1,229 @@
+#!/bin/bash
```
Comment (nirs) [inline hack/metrodr-setup.sh:1]: Other scripts use sh - bash is better but do we really want to maintain both sh and bash scripts?

I would remove both bash and sh and convert use python.

File: hack/metrodr-setup.sh [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+function check_for_command() {
+	if ! command -v "$1" >/dev/null
+	then
+		echo "command $1 not found"
+		exit 1
+	fi
+}
+
+function externalCluster_setFSID() {
```
Comment (nirs) [inline hack/metrodr-setup.sh:82]: Inconsistent style, rest of code in hack seems to use lower_case.

File: hack/metrodr-setup.sh [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+        declare rook_tools_pod
+        rook_tools_pod=$(kubectl --context "$1" -n "$2" get pods -o name | grep tools)
```
Comment (nirs) [inline hack/metrodr-setup.sh:84]: Using $1 and $2 make this code very hard to understand. Please do not add this in new code. This should be eliminated entirely in this project.

**Comment by raghavendra-talur** [comment]
@nirs I have made some changes to this draft PR as it is something I am using for my current sync dr testing. I have addressed some of your comments as part of the changes but I don't intend to bring this PR out of draft status.

Please continue working on your changes using drenv. My goal is to have this working so that we can use it as a reference when your drenv code is merged to enable sync dr in that.

**Pull request closed without merging**